tianshou/data/buffer.py

```diff
@@ -181,20 +79,95 @@ def __setstate__(self, state: Dict[str, Any]) -> None:
         We need it because pickling buffer does not work out-of-the-box
         ("buffer.__getattr__" is customized).
         """
-        self._indices = np.arange(state["_maxsize"])
         self.__dict__.update(state)
+        # compatible with previous version HDF5
+        self._indices = np.arange(self.maxsize)
 
-    def __getstate__(self) -> dict:
-        exclude = {"_indices"}
-        state = {k: v for k, v in self.__dict__.items() if k not in exclude}
-        return state
+    def __getstate__(self) -> Dict[str, Any]:
+        return self.__dict__
```

Comment:
Why not to get rid of this method completely since it is the [default](https://docs.python.org/3/library/pickle.html#object.__getstate__).